app/src/components/market/common/history_section/historySelect.tsx

```diff
@@ -0,0 +1,342 @@
+
+import { useConnectedWeb3Context, useContracts } from '../../../../hooks'
+import {
+  FpmmTradeDataType,
+  useGraphFpmmTransactionsFromQuestion,
+} from '../../../../hooks/useGraphFpmmTransactionsFromQuestion'
+import { CPKService } from '../../../../services'
+import { calcPrice, calcSellAmountInCollateral, calculateSharesBought, formatBigNumber } from '../../../../util/tools'
+import { HistoricData, Period } from '../../../../util/types'
+import { Button, ButtonSelectable } from '../../../button'
+import { Dropdown, DropdownPosition } from '../../../common/form/dropdown'
+import { InlineLoading } from '../../../loading'
+import { HistoryChart } from '../history_chart'
+import { HistoryTable } from '../history_table'
+
+const commonWrapperCSS = css`
+  border-top: 1px solid ${props => props.theme.borders.borderDisabled};
+  margin-left: -${props => props.theme.cards.paddingHorizontal};
+  margin-right: -${props => props.theme.cards.paddingHorizontal};
+  width: auto;
+`
+const DropdownMenu = styled(Dropdown)`
+  margin-left: auto;
+  width: 33%;
+`
+
+const NoData = styled.div`
+  ${commonWrapperCSS};
+  align-items: center;
+  color: ${props => props.theme.colors.textColorDarker};
+  display: flex;
+  font-size: 15px;
+  font-weight: 400;
+  height: 340px;
+  justify-content: center;
+  letter-spacing: 0.4px;
+  line-height: 1.3;
+  padding-left: ${props => props.theme.cards.paddingHorizontal};
+  padding-right: ${props => props.theme.cards.paddingHorizontal};
+`
+
+const CustomInlineLoading = styled(InlineLoading)`
+  ${commonWrapperCSS};
+  height: 340px;
+`
+
+const ChartWrapper = styled.div`
+  ${commonWrapperCSS}
+`
+
+const TitleWrapper = styled.div`
+  align-items: center;
+  border-bottom: 1px solid ${props => props.theme.borders.borderDisabled};
+  display: flex;
+  margin: 0;
+  padding: 20px 24px;
+`
+
+const ButtonsWrapper = styled.div`
+  align-items: center;
+  display: flex;
+  flex: 1;
+  justify-content: space-between;
+  margin-left: 80px;
+`
+
+const SelectWrapper = styled.div`
+  display: flex;
+`
+
+type Props = {
+  holdingSeries: Maybe<HistoricData>
+  onChange: (s: Period) => void
+  options: Period[]
+  outcomes: string[]
+  value: Period
+  currency: string
+  marketMakerAddress: string
+  decimals: number
+  fee: BigNumber
+}
+
+const ButtonSelectableStyled = styled(ButtonSelectable)<{ active?: boolean }>`
+  color: ${props => (props.active ? props.theme.colors.primary : props.theme.colors.clickable)};
+  font-weight: 500;
+  margin-left: 5px;
+
+  &:first-child {
+    margin-left: 0;
+  }
+`
+const ButtonSelect = styled(Button)<{ active: boolean }>`
+  margin-right: 10px;
+  padding: 20px 15px;
+  ${props => (props.active ? `border-color:${props.theme.colors.borderColorDark}` : '')};
+`
+
+const timestampToDate = (timestamp: number, value: string) => {
+  const ts = moment(timestamp * 1000)
+  if (value === '1D' || value === '1H') return ts.format('HH:mm')
+
+  return ts.format('MMM D')
+}
```

Comment:
Hmm yeah that would be good but it would require extra evaluation step before formatting since as you can see it has two return statments.
![Screenshot 2021-01-12 at 17 08 25](https://user-images.githubusercontent.com/33226956/104340260-c9d90e00-54f8-11eb-9220-b35984035040.png)